incubator/validator-avro/src/main/java/io/aklivity/zilla/runtime/validator/avro/AvroValidator.java

```diff
@@ -114,14 +116,15 @@ public DirectBuffer read(
             convertResponse(data, value, payloadBytes, schema);
         }
 
-        return value;
+        return value.capacity() == 0 ? -1 : next.applyAsInt(value, index, value.capacity());
```

Comment:
Now that we don't have to return the `value`, the allocations for `value`, `payloadBytes`, `byteBuf`, `valBytes` can be eliminated, or reuse shared instances as necessary that are only needed on stack during method invocation, so no chance of concurrent reuse.

Reply:
Note the return value here should be how much of `length` was processed, whereas the return value of `next` is an indication of how much of what was passed into `next` was processed.